src/service/url-replacements-impl.js

```diff
@@ -128,17 +128,19 @@ export class GlobalVariableSource extends VariableSource {
     });
 
     // Returns the canonical URL for this AMP document.
-    this.set('CANONICAL_URL', this.getDocInfoUrl_('canonicalUrl'));
+    this.set('CANONICAL_URL', () => this.getDocInfo_().canonicalUrl);
 
     // Returns the host of the canonical URL for this AMP document.
-    this.set('CANONICAL_HOST', this.getDocInfoUrl_('canonicalUrl', 'host'));
+    this.set('CANONICAL_HOST', () =>
+      parseUrlDeprecated(this.getDocInfo_().canonicalUrl)['host']);
```

Comment:
Can we not use the `parseUrlDeprecated` method, but use the url service instead.

Reply:
Done. I assumed it was still in use for a reason, but I guess it was an oversight 🤷‍♂️

Reply:
Looks like this change broke everything. :( Working on fix.